### PR TITLE
Unpin Haystack dependencies

### PR DESCRIPTION
**Pull request by kylesayrs**
After encountering issues attempting to install Haystack and its dependencies with different versions of python, it was decided to unpin the specific versions and instead match the dependency versioning provided by Haystack 1.4.0.

Installation was tested and successful with python3.8 and python3.9. The readme docs will specify that deepsparse[haystack] should only be installed with python3.8 pending better torch support in sparseml for other python versions.

### CODE REVIEW

**Comment by dhuangnm** [comment]
Just wondering do we really need that many number of dependencies?

**Comment by kylesayrs** [comment]
@dhuangnm These are all the dependencies that are required for `haystack[all]`. Theoretically we could get by with the core haystack dependencies, but then users would have to reinstall haystack to get features like Faiss and Weaviate.

**Comment by dhuangnm** [comment]
I see, can we also get the cherry-pick PR up and landed today so we can pick up the fix in the engine?

**Review by bfineran — APPROVED** [review]

File: src/deepsparse/transformers/haystack/haystack_reqs.txt [hunk 1]
```diff
@@ -8,280 +8,76 @@
...
-alembic==1.8.0
-appdirs==1.4.4
-astroid==2.11.7
-asttokens==2.0.5
-async-generator==1.10
-async-timeout==4.0.2
-attrs==21.4.0
-audioread==2.1.9
+importlib-metadata
+#torch>1.9,<1.11
```
Comment (rahul-tuli) [inline src/deepsparse/transformers/haystack/haystack_reqs.txt:19]: Nit: remove

File: src/deepsparse/transformers/haystack/haystack_reqs.txt [hunk 2]
```diff
@@ -8,280 +8,76 @@
...
+torch==1.9.1
+requests
+pydantic
+#transformers==4.13.0
```
Comment (rahul-tuli) [inline src/deepsparse/transformers/haystack/haystack_reqs.txt:23]: Nit: remove

**Review by rahul-tuli — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**